### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by lucifercr07**
Unit tests being added for the following files:
- cookie.js
- header.js
- request-auth.js
- item.js

Increased coverage from 96.54% to 98.15%.

### CODE REVIEW

File: test/unit/item.test.js [hunk 1]
```diff
@@ -466,6 +466,15 @@ describe('Item', function () {
             item.unsetProtocolProfileBehavior('keyName');
             expect(item.toJSON()).to.have.property('protocolProfileBehavior').that.is.empty;
         });
+
+        it('should not unset protocolProfileBehavior from an Item', function () {
```
Comment (codenirvana) [inline test/unit/item.test.js:470]: Misleading title. It says,
> unsetProtocolProfileBehavior should not unset protocolProfileBehavior from an Item

File: test/unit/request-auth.test.js [hunk 2]
```diff
@@ -73,6 +73,55 @@ describe('RequestAuth', function () {
         });
     });
 
+    it('should be able to construct with multiple options', function () {
```
Comment (codenirvana) [inline test/unit/request-auth.test.js:76]: Not sure what this test is about 😕

File: test/unit/request-auth.test.js [hunk 3]
```diff
@@ -73,6 +73,55 @@ describe('RequestAuth', function () {
...
+    it('should be able to return the parameters of the selected auth type', function () {
+        var auth = new RequestAuth({
+            noauth: {
+                foo: 'bar'
+            },
+            type: 'noauth'
+        });
+
+        auth.use('noauth', []);
+        expect(auth.type).to.equal('noauth');
```
Comment (codenirvana) [inline test/unit/request-auth.test.js:103]: Why are these assertions necessary?

File: test/unit/request-auth.test.js [hunk 4]
```diff
@@ -73,6 +73,55 @@ describe('RequestAuth', function () {
...
+        expect(auth.parameters().toObject()).to.eql({
+            foo: 'bar'
+        });
+    });
+
+    it('should return if options not present', function () {
```
Comment (codenirvana) [inline test/unit/request-auth.test.js:109]: 🤔

File: test/unit/header.test.js [hunk 5]
```diff
@@ -229,4 +238,14 @@ describe('Header', function () {
             expect(Header.isHeader()).to.be.false;
         });
     });
+
+    describe('toString', function () {
+        // eslint-disable-next-line max-len
+        var rawHeader = 'Content-Type: application/json\nAuthorization: Hawk id="dh37fgj492je", ts="1448549987", nonce="eOJZCd", mac="O2TFlvAlMvKVSKOzc6XkfU6+5285k5p3m5dAjxumo2k="';
+
+        it('should the header to a single header string', function () {
```
Comment (codenirvana) [inline test/unit/header.test.js:246]: > toString should the header to a single header string

File: test/unit/cookie.test.js [hunk 6]
```diff
@@ -194,4 +194,133 @@ describe('Cookie', function () {
...
+        // @todo: check why the test report doesn't say 230 line covered cookie.js
+        it('should handle malformed URI sequence', function () {
+            expect((new Cookie({
+                name: 'blah',
+                value: '%E0%A4%A'
+            })).valueOf()).to.eql('%E0%A4%A');
+        });
+    });
+});
+
```
Comment (codenirvana) [inline test/unit/cookie.test.js:208]: Why all the tests below are not nested under `Cookie` describe block?

**Review by codenirvana — CHANGES_REQUESTED** [review]
Revisit the test titles.

File: test/unit/request-auth.test.js (hunk 2 above)
Comment (lucifercr07) [reply test/unit/request-auth.test.js:76]: Removed the test had no impact on coverage.

File: test/unit/request-auth.test.js (hunk 3 above)
Comment (lucifercr07) [reply test/unit/request-auth.test.js:99]: Removed the assertions were not required strictly.

File: test/unit/cookie.test.js [hunk 7]
```diff
@@ -194,4 +194,142 @@ describe('Cookie', function () {
...
+                    value: 'GA1.2.113558537.1435817423',
+                    maxAge: 1502442248,
+                    extensions: [{
+                        key: 'Priority',
+                        value: 'HIGH'
+                    }]
+                },
+                cookie = new Cookie(rawCookie);
+            expect(cookie.toString())
+                .equals('_ga=GA1.2.113558537.1435817423; Max-Age=1502442248; [object Object]');
```
Comment (codenirvana) [inline test/unit/cookie.test.js:332]: What's `[object Object]`?

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: test/unit/cookie.test.js [hunk 8]
```diff
@@ -194,4 +194,142 @@ describe('Cookie', function () {
             })).valueOf()).to.eql('this is a cookie value');
         });
     });
+
+    describe('valueOf', function () {
+        // @todo: check why the test report doesn't say 230 line covered cookie.js
```
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/cookie.test.js:199]: The error is not getting thrown, even though the URI is malformed: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent

File: test/unit/cookie.test.js (hunk 7 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js:332]: with extensions as part of cookie `toString` converts to [object Object] is it valid, thus need to do this comparison

File: test/unit/request-auth.test.js [hunk 9]
```diff
@@ -73,6 +73,61 @@ describe('RequestAuth', function () {
...
+            }),
+            options = {
+                basic: {
+                    username: 'u', password: 'p'
+                },
+                type: 'noauth' };
+
+        auth.use('noauth', []);
+        auth.update(options, 'type');
+        expect(auth.parameters().toObject()).to.eql({
```
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/request-auth.test.js:144]: Fixed, added the code change to return if invalid auth

File: test/unit/request-auth.test.js [hunk 10]
```diff
@@ -73,6 +73,61 @@ describe('RequestAuth', function () {
         });
     });
```
[the remark this answers is not in the document]
Comment (lucifercr07) [reply test/unit/request-auth.test.js:75]: Fixed

File: test/unit/content-info.test.js [hunk 11]
```diff
@@ -174,6 +174,28 @@ describe('contentInfo module', function () {
             });
     });
 
+    it('Should return if invalid charset type specified', function () {
```
Comment (codenirvana) [inline test/unit/content-info.test.js:177]: return what?

File: test/unit/cookie.test.js [hunk 12]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+        it('should handle malformed URI sequence', function () {
+            expect((new Cookie({
+                name: 'blah',
+                value: '%E0%A4%A'
+            })).valueOf()).to.eql('%E0%A4%A');
+        });
+    });
+
+    describe('unparseSingle', function () {
+        var rawCookie = 'GAPS=lol;Path=/;Expires=Sun, 04-Feb-2018 14:18:27 GMT;Secure;HttpOnly;Priority=HIGH';
```
Comment (codenirvana) [inline test/unit/cookie.test.js:208]: why is this variable declared outside `it` block?

File: test/unit/cookie.test.js [hunk 13]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+        it('should be unparsed properly in case of no cookie name', function () {
+            var unparsedSingle = Cookie.unparseSingle(rawCookie);
+            expect(unparsedSingle).equals('');
```
Comment (codenirvana) [inline test/unit/cookie.test.js:211]: From the test title, I don't think this is asserting the right thing.

File: test/unit/cookie.test.js [hunk 14]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+                    session: false,
+                    value: 'GA1.2.113558537.1435817423'
+                },
+                unparsedSingle = Cookie.unparseSingle(rawCookie);
+            expect(unparsedSingle).equals(rawCookie.name + '=' + rawCookie.value);
+        });
+    });
+
+    describe('unparse', function () {
+        it('should return empty string if not array of cookies', function () {
```
Comment (codenirvana) [inline test/unit/cookie.test.js:232]:
```suggestion
        it('should return an empty string for non-array arguments', function () {
```
and test with multiple types.

File: test/unit/cookie.test.js [hunk 15]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+                    httpOnly: false,
+                    key: '_ga',
+                    path: '/',
+                    secure: false,
+                    session: false,
+                    value: 'GA1.2.113558537.1435817423'
+                },
+                cookieArray = [];
+            cookieArray.push(rawCookie);
+            // eslint-disable-next-line one-var
```
Comment (codenirvana) [inline test/unit/cookie.test.js:250]: 🤦🏻‍♂️

File: test/unit/cookie.test.js [hunk 16]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+            var unparse = Cookie.unparse(cookieArray);
+            expect(unparse).equals(cookieArray.map(Cookie.unparseSingle).join('; '));
```
Comment (codenirvana) [inline test/unit/cookie.test.js:252]: why?

File: test/unit/cookie.test.js [hunk 17]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+        });
+    });
+
+    describe('stringify', function () {
+        var rawCookie = {
```
Comment (codenirvana) [inline test/unit/cookie.test.js:257]: can we avoid shared globals, please 🙏🏻

File: test/unit/cookie.test.js [hunk 18]
```diff
@@ -194,4 +194,161 @@ describe('Cookie', function () {
...
+        });
+    });
+
+    describe('toString', function () {
+        it('should return single Set-Cookie header string with key value', function () {
+            var rawCookie = {
+                    key: '_ga',
+                    value: 'GA1.2.113558537.1435817423'
+                },
+                cookie = new Cookie(rawCookie);
```
Comment (codenirvana) [inline test/unit/cookie.test.js:282]: `rawCookie` is used only once, the object can be passed here directly.

File: lib/collection/cookie.js [hunk 19]
```diff
@@ -270,7 +273,13 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
...
         if (this.extensions) {
             this.extensions.forEach(function (ext) {
-                str += '; ' + ext;
+                try {
+                    extensionString = JSON.stringify(ext);
+                }
+                catch (e) {
+                    return str;
+                }
+                str += '; ' + extensionString;
```
Comment (codenirvana) [inline lib/collection/cookie.js:282]: `extensions` string representation is not a stringified object.

File: test/unit/item.test.js [hunk 20]
```diff
@@ -466,6 +466,15 @@ describe('Item', function () {
             item.unsetProtocolProfileBehavior('keyName');
             expect(item.toJSON()).to.have.property('protocolProfileBehavior').that.is.empty;
         });
+
+        it('should not unset protocolProfileBehavior from an Item incase key is not of type string', function () {
+            var item = new Item({
+                protocolProfileBehavior: { keyName: 'null' }
```
Comment (codenirvana) [inline test/unit/item.test.js:472]: https://github.com/postmanlabs/postman-collection/pull/1084#discussion_r479958392

I meant to set the `key` to null not set the value as null.

**Review by codenirvana — CHANGES_REQUESTED** [review]
Never force-push once the review is started, I can't refer to my past comments anymore.

**Comment by codenirvana** [comment]
No point in reviewing this PR after force-push, please create a new PR.

**Pull request closed without merging**